lib/resource_safety.ts

```diff
@@ -13,14 +13,26 @@
  * permissions and limitations under the License.
  */
 
- export interface ResourceSafe {
-     close(): void;
- }
+/*
+ * If you have a resource that you want typescript to enforce close is implemented
+ * and/or you want to use the below 'using' function, then implement this interface.
+ */ 
+export interface ResourceSafe {
+    close(): void;
+}
 
- export async function using<T extends ResourceSafe>(resource : T, func: (resource: T) => void) {
-     try {
-         await func(resource);
-     } finally {
-         resource.close();
-     }
- }
\ No newline at end of file
+/*
+ * Use this function to create a resource in an async context. This will make sure the 
+ * resources are cleaned up before returning.
+ * 
+ * example:   await using(res = new SomeResource()) {
```

Comment:
this isn't the right syntax, it should take the lambda as the second argument in ```using()```